src/terminal/adapter/DispatchTypes.hpp

```diff
@@ -52,6 +57,17 @@ namespace Microsoft::Console::VirtualTerminal
         uint64_t _value;
     };
 
+    inline std::wostream& operator<<(std::wostream& out, const VTID& id)
+    {
+        uint64_t value = id;
+        do
+        {
+            out << gsl::narrow_cast<char>(value & 0xFF);
+            value >>= CHAR_BIT;
+        } while (value);
+        return out;
+    }

```

Comment:
It'd be possible to turn `VTID` into a:
```cpp
union VTID
{
private:
    uint64_t _value = 0;
    char _string[8];
};
```
and it'd be entirely valid C++. That way we could get the string out of it in constant time and simplify both string related functions to a simple `std::copy_n` for `FromString` and `return &_string[0]` for the `ToString` function. All we need to do to make this work is to ensure that `_string[7]` is 0 in the 2 constructors this class has, which I feel like is a feasible modification to do.

Reply:
Yeah, I think this is good idea. I'll give it a try.

Reply:
In the end I just used your union trick to get a more efficient `ToString` and left the rest more or less as it was (other than making sure there was always a null terminator).

I had initially thought I could make the `ToString` return a `char*`, but the `fmt::format` call couldn't cope with that (I'm assuming because of the mismatch between `char` and `wchar_t`). It was happy with a `string_view` though.

It's maybe not perfect, but I think this is an improvement on the initial implementation.